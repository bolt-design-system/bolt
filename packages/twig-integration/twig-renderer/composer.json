{
  "name": "bolt-design-system/twig-renderer",
  "type": "library",
  "description": "Twig rendering service used by the Bolt Design System.",
  "version": "2.20.0",
  "license": "MIT",
  "repositories": [
    {
      "type": "path",
	    "url": "../twig-extensions-shared"
	  },
    {
      "type": "path",
      "url": "../twig-extensions-compat"
    }
  ],
  "require": {
    "bolt-design-system/core-php": "*",
    "bolt-design-system/drupal-twig-extensions": "*",
    "twig/twig": "^1.42.5",
    "webmozart/path-util": "*",
    "symfony/finder": "*"
  },
  "scripts": {
    "setup": "@composer install --no-interaction --prefer-dist --no-progress"
  }
}
