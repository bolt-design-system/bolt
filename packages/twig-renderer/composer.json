{
  "name": "bolt-design-system/twig-renderer",
  "type": "library",
  "version": "2.3.0",
  "license": "MIT",
  "require": {
    "bolt-design-system/core-php": "^2.3.0",
    "bolt-design-system/drupal-twig-extensions": "*",
    "twig/twig": "^1.38.2",
    "webmozart/path-util": "^2.3",
    "symfony/finder": "*"
  },
  "scripts": {
    "setup": "@composer install --no-interaction --prefer-dist --no-progress"
  }
}
