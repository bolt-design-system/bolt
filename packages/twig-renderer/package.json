{
  "name": "@bolt/twig-renderer",
  "version": "2.7.0",
  "license": "MIT",
  "description": "Renders Twig templates using an on-the-fly PHP rendering service.",
  "main": "twig-renderer.js",
  "repository": "https://github.com/bolt-design-system/bolt/tree/master/packages/twig-renderer",
  "keywords": [
    "design system",
    "bolt design system",
    "twig",
    "php"
  ],
  "maintainers": [
    {
      "name": "Salem Ghoweri",
      "email": "me@salemghoweri.com",
      "web": "https://github.com/sghoweri"
    }
  ],
  "dependencies": {
    "@basalt/twig-renderer": "^0.12.1",
    "@bolt/build-utils": "^2.7.0",
    "@bolt/drupal-twig-extensions": "^2.7.0",
    "sleep-promise": "^8.0.1"
  },
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "setup": "composer run setup"
  },
  "schema": "utils/config.schema.yml",
  "bugs": "https://github.com/bolt-design-system/bolt/issues",
  "homepage": "https://boltdesignsystem.com",
  "gitHead": "3d198feea5f73c63913a956a2a077e229ad8d462"
}
