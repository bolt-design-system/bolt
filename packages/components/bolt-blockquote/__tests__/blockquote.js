import { render } from '@bolt/twig-renderer';
import { fixture as html } from '@open-wc/testing-helpers';

const { readYamlFileSync } = require('@bolt/build-tools/utils/yaml');
const { join } = require('path');
const schema = readYamlFileSync(join(__dirname, '../blockquote.schema.yml'));
const { tag } = schema.properties;

const timeout = 90000;

describe('button', () => {
  let page;

  beforeEach(async () => {
    page = await global.__BROWSER__.newPage();
    await page.goto('http://127.0.0.1:4444/', {
      timeout: 0,
      waitLoad: true,
      waitNetworkIdle: true, // defaults to false
    });
  }, timeout);

  test('Basic usage', async () => {
    const results = await render(
      '@bolt-components-blockquote/blockquote.twig',
      {
        author: {
          name: 'Michelangelo di Lodovico Buonarroti Simoni',
          title: 'Renaissance Artist',
        },
        content:
          '<p>The greater danger for most of us lies not in setting our aim too high and falling short; but in setting our aim too low, and achieving our mark.</p>',
      },
    );
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  test('Default <bolt-blockquote> with Shadow DOM renders', async function() {
    const defaultBlockquoteOuter = await page.evaluate(() => {
      const blockquote = document.createElement('bolt-blockquote');
      blockquote.setAttribute(
        'author-name',
        'Michelangelo di Lodovico Buonarroti Simoni',
      );
      blockquote.setAttribute('author-title', 'Renaissance Artist');
      blockquote.setAttribute('author-image', '/fixtures/500x500.jpg');
      blockquote.innerHTML = `<img slot="logo" src="/fixtures/logo-paypal.svg" alt="PayPal Logo">
        <p>The greater danger for most of us lies not in setting our aim too high and falling short...</p>
        <p>In fact, the greater danger is setting our aim too low and achieving our mark.</p>`;
      document.body.appendChild(blockquote);
      blockquote.updated();
      return blockquote.outerHTML;
    });

    const renderedHTML = await html(defaultBlockquoteOuter);

    const image = await page.screenshot();

    expect(image).toMatchImageSnapshot({
      failureThreshold: '0.03',
      failureThresholdType: 'percent',
    });

    expect(renderedHTML).toMatchSnapshot();
  });

  test('Default <bolt-blockquote> w/o Shadow DOM renders', async function() {
    const renderedBlockquoteHTML = await page.evaluate(() => {
      const blockquote = document.createElement('bolt-blockquote');
      blockquote.setAttribute(
        'author-name',
        'Michelangelo di Lodovico Buonarroti Simoni',
      );
      blockquote.setAttribute('author-title', 'Renaissance Artist');
      blockquote.setAttribute('author-image', '/fixtures/500x500.jpg');
      blockquote.innerHTML = `<img slot="logo" src="/fixtures/logo-paypal.svg" alt="PayPal Logo">
        <p>The greater danger for most of us lies not in setting our aim too high and falling short...</p>
        <p>In fact, the greater danger is setting our aim too low and achieving our mark.</p>`;
      document.body.appendChild(blockquote);
      blockquote.useShadow = false;
      blockquote.updated();
      return blockquote.outerHTML;
    });

    const renderedHTML = await html(renderedBlockquoteHTML);

    expect(
      renderedHTML
        .querySelector('.c-bolt-blockquote')
        .classList.contains('c-bolt-blockquote--xlarge'),
    ).toBe(true);

    const image = await page.screenshot();

    expect(image).toMatchImageSnapshot({
      failureThreshold: '0.03',
      failureThresholdType: 'percent',
    });

    expect(renderedHTML).toMatchSnapshot();
  });

  test('Default <bolt-blockquote> inside a <form> renders', async function() {
    const renderedBlockquoteHTML = await page.evaluate(() => {
      const form = document.createElement('form');
      const blockquote = document.createElement('bolt-blockquote');
      blockquote.setAttribute(
        'author-name',
        'Michelangelo di Lodovico Buonarroti Simoni',
      );
      blockquote.setAttribute('author-title', 'Renaissance Artist');
      blockquote.setAttribute('author-image', '/fixtures/500x500.jpg');
      blockquote.innerHTML = `<img slot="logo" src="/fixtures/logo-paypal.svg" alt="PayPal Logo">
        <p>The greater danger for most of us lies not in setting our aim too high and falling short...</p>
        <p>In fact, the greater danger is setting our aim too low and achieving our mark.</p>`;
      document.body.appendChild(form);
      form.appendChild(blockquote);
      blockquote.updated();
      return blockquote.innerHTML;
    });

    const renderedHTML = await html(renderedBlockquoteHTML);

    const image = await page.screenshot();

    expect(image).toMatchImageSnapshot({
      failureThreshold: '0.03',
      failureThresholdType: 'percent',
    });

    expect(renderedHTML).toMatchSnapshot();
  });
});
