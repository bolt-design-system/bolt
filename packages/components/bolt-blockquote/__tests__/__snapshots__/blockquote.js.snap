// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button Basic usage 1`] = `

<bolt-blockquote>
  <blockquote class="c-bolt-blockquote c-bolt-blockquote--xlarge c-bolt-blockquote--align-items-start c-bolt-blockquote--bordered-vertical"
              is="shadow-root"
  >
    <replace-with-grandchildren class="c-bolt-blockquote__quote">
      <div class="c-bolt-text c-bolt-text--semibold c-bolt-text--normal c-bolt-text--xlarge">
        <p>
          The greater danger for most of us lies not in setting our aim too high and falling short; but in setting our aim too low, and achieving our mark.
        </p>
      </div>
    </replace-with-grandchildren>
    <replace-with-grandchildren>
      <footer class="c-bolt-blockquote__footer">
        <replace-with-children class="c-bolt-blockquote__footer-item">
          <cite slot="author-name"
                class="c-bolt-text c-bolt-text--bold c-bolt-text--normal c-bolt-text--xsmall"
          >
            Michelangelo di Lodovico Buonarroti Simoni
          </cite>
        </replace-with-children>
        <replace-with-children class="c-bolt-blockquote__footer-item">
          <cite slot="author-title"
                class="c-bolt-text c-bolt-text--regular c-bolt-text--normal c-bolt-text--xsmall"
          >
            Renaissance Artist
          </cite>
        </replace-with-children>
      </footer>
    </replace-with-grandchildren>
  </blockquote>
</bolt-blockquote>

`;

exports[`button Default <bolt-blockquote> inside a <form> renders 2`] = `
<blockquote
  class="c-bolt-blockquote c-bolt-blockquote--xlarge c-bolt-blockquote--align-items-start c-bolt-blockquote--bordered-vertical"
  is="shadow-root"
  style=""
>
  
        
              
  <div
    class="c-bolt-blockquote__logo"
    style=""
  >
    
                
    <!---->
    <!---->
    <img
      alt="PayPal Logo"
      slot="logo"
      src="/fixtures/logo-paypal.svg"
      style=""
    >
      <!---->
      <!---->
      
              
    </img>
  </div>
  
            
        
  <div
    class="c-bolt-blockquote__quote"
    style=""
  >
    
          
    <div
      class="c-bolt-text-v2 c-bolt-text-v2--block c-bolt-text-v2--body c-bolt-text-v2--font-size-xlarge c-bolt-text-v2--font-weight-semibold c-bolt-text-v2--font-style-regular c-bolt-text-v2--color-theme-headline c-bolt-text-v2--letter-spacing-regular c-bolt-text-v2--align-inherit c-bolt-text-v2--text-transform-regular c-bolt-text-v2--line-height-regular c-bolt-text-v2--opacity-100"
      style=""
    >
      
            
      <!---->
      <!---->
      
        
      <!---->
      <p
        class="is-first-child"
        style=""
      >
        The greater danger for most of us lies not in setting our aim too high and falling short...
      </p>
      <!---->
      
        
      <!---->
      <p
        class="is-last-child"
        style=""
      >
        In fact, the greater danger is setting our aim too low and achieving our mark.
      </p>
      <!---->
      <!---->
      
          
    </div>
    
        
  </div>
  
        
              
  <footer
    class="c-bolt-blockquote__footer"
    style=""
  >
    
                
    <!---->
    
                    
    <div
      class="c-bolt-blockquote__footer-item"
      style=""
    >
      
                      
      
      <div
        class="c-bolt-blockquote__image"
        style=""
      >
        
        
              
        <img
          alt="Renaissance Artist"
          src="/fixtures/500x500.jpg"
          style=""
        >
          
            
      
        </img>
      </div>
      
    
                    
    </div>
    
                  
    <!---->
    
                    
    <div
      class="c-bolt-blockquote__footer-item"
      style=""
    >
      
                      
      
      <cite
        class="c-bolt-text-v2 c-bolt-text-v2--block c-bolt-text-v2--body c-bolt-text-v2--font-size-xsmall c-bolt-text-v2--font-weight-bold c-bolt-text-v2--font-style-regular c-bolt-text-v2--color-theme-headline c-bolt-text-v2--letter-spacing-regular c-bolt-text-v2--align-inherit c-bolt-text-v2--text-transform-regular c-bolt-text-v2--line-height-regular c-bolt-text-v2--opacity-100"
        style=""
      >
        
        Michelangelo di Lodovico Buonarroti Simoni
      
      </cite>
      
    
                    
    </div>
    
                  
    <!---->
    
                    
    <div
      class="c-bolt-blockquote__footer-item"
      style=""
    >
      
                      
      
      <cite
        class="c-bolt-text-v2 c-bolt-text-v2--block c-bolt-text-v2--body c-bolt-text-v2--font-size-xsmall c-bolt-text-v2--font-weight-regular c-bolt-text-v2--font-style-regular c-bolt-text-v2--color-theme-headline c-bolt-text-v2--letter-spacing-regular c-bolt-text-v2--align-inherit c-bolt-text-v2--text-transform-regular c-bolt-text-v2--line-height-regular c-bolt-text-v2--opacity-100"
        style=""
      >
        
        Renaissance Artist
      
      </cite>
      
    
                    
    </div>
    
                  
    <!---->
    
              
  </footer>
  
            
      
</blockquote>
`;

exports[`button Default <bolt-blockquote> w/o Shadow DOM renders 2`] = `
<bolt-blockquote
  author-image="/fixtures/500x500.jpg"
  author-name="Michelangelo di Lodovico Buonarroti Simoni"
  author-title="Renaissance Artist"
  style=""
>
  <!---->
  
      
      
  <blockquote
    class="c-bolt-blockquote c-bolt-blockquote--xlarge c-bolt-blockquote--align-items-start c-bolt-blockquote--bordered-vertical"
    is="shadow-root"
    style=""
  >
    
        
              
    <div
      class="c-bolt-blockquote__logo"
      style=""
    >
      
                
      <!---->
      <!---->
      <img
        alt="PayPal Logo"
        slot="logo"
        src="/fixtures/logo-paypal.svg"
        style=""
      >
        <!---->
        <!---->
        
              
      </img>
    </div>
    
            
        
    <div
      class="c-bolt-blockquote__quote"
      style=""
    >
      
          
      <div
        class="c-bolt-text-v2 c-bolt-text-v2--block c-bolt-text-v2--body c-bolt-text-v2--font-size-xlarge c-bolt-text-v2--font-weight-semibold c-bolt-text-v2--font-style-regular c-bolt-text-v2--color-theme-headline c-bolt-text-v2--letter-spacing-regular c-bolt-text-v2--align-inherit c-bolt-text-v2--text-transform-regular c-bolt-text-v2--line-height-regular c-bolt-text-v2--opacity-100"
        style=""
      >
        
            
        <!---->
        <!---->
        
        
        <!---->
        <p
          class="is-first-child"
          style=""
        >
          The greater danger for most of us lies not in setting our aim too high and falling short...
        </p>
        <!---->
        
        
        <!---->
        <p
          class="is-last-child"
          style=""
        >
          In fact, the greater danger is setting our aim too low and achieving our mark.
        </p>
        <!---->
        <!---->
        
          
      </div>
      
        
    </div>
    
        
              
    <footer
      class="c-bolt-blockquote__footer"
      style=""
    >
      
                
      <!---->
      
                    
      <div
        class="c-bolt-blockquote__footer-item"
        style=""
      >
        
                      
      
        <div
          class="c-bolt-blockquote__image"
          style=""
        >
          
        
              
          <img
            alt="Renaissance Artist"
            src="/fixtures/500x500.jpg"
            style=""
          >
            
            
      
          </img>
        </div>
        
    
                    
      </div>
      
                  
      <!---->
      
                    
      <div
        class="c-bolt-blockquote__footer-item"
        style=""
      >
        
                      
      
        <cite
          class="c-bolt-text-v2 c-bolt-text-v2--block c-bolt-text-v2--body c-bolt-text-v2--font-size-xsmall c-bolt-text-v2--font-weight-bold c-bolt-text-v2--font-style-regular c-bolt-text-v2--color-theme-headline c-bolt-text-v2--letter-spacing-regular c-bolt-text-v2--align-inherit c-bolt-text-v2--text-transform-regular c-bolt-text-v2--line-height-regular c-bolt-text-v2--opacity-100"
          style=""
        >
          
        Michelangelo di Lodovico Buonarroti Simoni
      
        </cite>
        
    
                    
      </div>
      
                  
      <!---->
      
                    
      <div
        class="c-bolt-blockquote__footer-item"
        style=""
      >
        
                      
      
        <cite
          class="c-bolt-text-v2 c-bolt-text-v2--block c-bolt-text-v2--body c-bolt-text-v2--font-size-xsmall c-bolt-text-v2--font-weight-regular c-bolt-text-v2--font-style-regular c-bolt-text-v2--color-theme-headline c-bolt-text-v2--letter-spacing-regular c-bolt-text-v2--align-inherit c-bolt-text-v2--text-transform-regular c-bolt-text-v2--line-height-regular c-bolt-text-v2--opacity-100"
          style=""
        >
          
        Renaissance Artist
      
        </cite>
        
    
                    
      </div>
      
                  
      <!---->
      
              
    </footer>
    
            
      
  </blockquote>
  
    
  <!---->
</bolt-blockquote>
`;

exports[`button Default <bolt-blockquote> with Shadow DOM renders 2`] = `
<bolt-blockquote
  author-image="/fixtures/500x500.jpg"
  author-name="Michelangelo di Lodovico Buonarroti Simoni"
  author-title="Renaissance Artist"
  style=""
>
  <img
    alt="PayPal Logo"
    slot="logo"
    src="/fixtures/logo-paypal.svg"
    style=""
  >
    
        
  </img>
  <p
    class="is-first-child"
    style=""
  >
    The greater danger for most of us lies not in setting our aim too high and falling short...
  </p>
  
        
  <p
    class="is-last-child"
    style=""
  >
    In fact, the greater danger is setting our aim too low and achieving our mark.
  </p>
</bolt-blockquote>
`;
